enforcer-parent/enforcer/src/main/java/org/wso2/choreo/connect/enforcer/security/jwt/APIKeyAuthenticator.java

```diff
@@ -193,6 +193,15 @@ public AuthenticationContext authenticate(RequestContext requestContext) throws
                         throw new APISecurityException(APIConstants.StatusCodes.SERVICE_UNAVAILABLE
                                 .getCode(), validationInfoDto.getValidationStatus(),
                                 GeneralErrorCodeConstants.API_BLOCKED_MESSAGE);
+                    } else if (APISecurityConstants.API_SUBSCRIPTION_BLOCKED == validationInfoDto
+                            .getValidationStatus()) {
+                        requestContext.getProperties().put(APIConstants.MessageFormat.ERROR_MESSAGE,
+                                APISecurityConstants.API_SUBSCRIPTION_BLOCKED_MESSAGE);
+                        requestContext.getProperties().put(APIConstants.MessageFormat.ERROR_DESCRIPTION,
+                                APISecurityConstants.API_SUBSCRIPTION_BLOCKED_DESCRIPTION);
+                        throw new APISecurityException(APIConstants.StatusCodes.SERVICE_UNAVAILABLE
```

Comment:
Status code should be 401. https://apim.docs.wso2.com/en/latest/troubleshooting/error-handling/